crates/ethereum/payload/src/lib.rs

```diff
@@ -124,16 +147,18 @@ where
 /// and configuration, this function creates a transaction payload. Returns
 /// a result indicating success with the payload or an error in case of failure.
 #[inline]
-pub fn default_ethereum_payload<EvmConfig, Pool, Client>(
+pub fn default_ethereum_payload<EvmConfig, Pool, Client, IteratorClosure>(
```

Comment:
let's just name this F
```suggestion
pub fn default_ethereum_payload<EvmConfig, Pool, Client, F>(
```